app/models/ems_refresh/save_inventory.rb

```diff
@@ -205,17 +205,23 @@ def save_guest_devices_inventory(hardware, hashes)
     store_ids_for_new_records(hardware.guest_devices, hashes, [:device_type, :uid_ems])
   end
 
-  def save_disks_inventory(hardware, hashes)
+  def save_disks_inventory(parent, hashes)
     return if hashes.nil?
 
     # Update the associated ids
     hashes.each do |h|
-      h[:storage_id]         = h.fetch_path(:storage, :id)
+      if h[:storage_id].nil?
+        h[:storage_id] = h.fetch_path(:storage, :id)
+      end
       h[:backing_id]         = h.fetch_path(:backing, :id)
       h[:storage_profile_id] = h.fetch_path(:storage_profile, :id)
     end
 
-    save_inventory_multi(hardware.disks, hashes, :use_association, [:controller_type, :location], nil, [:storage, :backing, :storage_profile])
+    filenames = hashes.collect { |h| h[:filename] }
+
+    all_disks = Disk.where(:filename => filenames)
+    save_inventory_multi_child(:disks, all_disks, parent, hashes, [:controller_type, :location, :device_type],
```

Comment:
@agrare @pkliczewski from the BZ I understand the problem only appears when we destroy Vm in RHEV, then we refresh. I assume there are 2 events, one internal when deleting VM in MIQ and one external from RHEV? 

 https://github.com/Ladas/manageiq/blob/559bd25f277d9ec169a82bdebd8d52f10cc00f82/db/fixtures/ae_datastore/ManageIQ/System/Event/EmsEvent/EVM.class/destroyvm_task_complete.yaml#L12-L12

https://github.com/Ladas/manageiq/blob/6bb24b2d0561b466f4f4e4bd23b90934211a1a0b/db/fixtures/ae_datastore/ManageIQ/System/Event/EmsEvent/RHEVM.class/user_remove_vm_finished.yaml#L12-L12

So this would seem to me like the cause, why the behavior is different, when removing from RHEV directly.